src/Raven.Server/Documents/Indexes/IndexNameFinder.cs

```diff
@@ -37,13 +37,13 @@ private static string FindName(string[] collections, IReadOnlyCollection<IndexFi
             
             var combinedFields = string.Join("And", fields.Select(x => IndexField.ReplaceInvalidCharactersInFieldName(x.Name)).OrderBy(x => x));
```

Comment:
@ayende This is only called when Auto index is created, so not for every query, only when there are no suitable indexes to satisfy the given one. I do not think that allocations here are problematic because of that.